### PR TITLE
VZ-9495 GH Actions workflow for BFS

### PR DESCRIPTION
**Pull request by stevshap**
This PR adds a new workflow to build Keycloak from source using GH actions

### CODE REVIEW

File: .github/workflows/BFS.yaml [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+      contents: read
+      packages: write
+      # This is used to complete the identity challenge
+      # with sigstore/fulcio when running outside of PRs.
+      id-token: write
+
+    steps:
+      # Get Keycloak version
+      - name: Get Keycloak version
+        run: echo "KEYCLOAK_VERSION=$(echo ${{ github.ref_name }} | sed 's:.*/::')" >> "$GITHUB_ENV"
```
Comment (agv426) [inline .github/workflows/BFS.yaml:45]: Will this work for BFS branch names? I don't think those will have a short name matching the keycloak version

File: .github/workflows/BFS.yaml [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+
+      # Checkout keycloak repo
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          ref: oracle/release/${{ env.KEYCLOAK_VERSION }}
```
Comment (agv426) [inline .github/workflows/BFS.yaml]: The branch here is the oracle/release/* branch but above we are also triggering when folks are using bfs/* branches. Seems like this won't work for BFS branches, but only for the oracle/release ones?

File: .github/workflows/BFS.yaml (hunk 1 above)
Comment (stevshap) [reply .github/workflows/BFS.yaml:45]: Branches must follow the naming convention bfs/***/KEYCLOAK_VERSION

Comment (stevshap) [reply .github/workflows/BFS.yaml:45]: Need branch name validation

File: .github/workflows/BFS.yaml (hunk 2 above)
Comment (stevshap) [reply .github/workflows/BFS.yaml]: Get rid of ref

File: .github/workflows/BFS.yaml (hunk 1 above)
Comment (stevshap) [reply .github/workflows/BFS.yaml:45]: examples include:
-  bfs/example/example/example/KEYCLOAK_VERSION
- bfs/example/KEYCLOAK_VERSION

**Review by agv426 — APPROVED** [review]

**Pull request merged**